modules/kull_m_string.c

```diff
@@ -133,6 +139,115 @@ PCWCHAR WPRINTF_TYPES[] =
 	L"\\x%02x",		// WPRINTF_HEX_PYTHON
 };
 
+
+BOOL md5(LPCVOID data, DWORD len, BYTE hash[MD5LEN]) {
+	BOOL ret = FALSE;
+	HCRYPTPROV hProv = 0;
+	HCRYPTHASH hHash = 0;
+
+	DWORD hash_len = 0;
+
+	// Get handle to the crypto provider
+	if (!CryptAcquireContext(&hProv,
+		NULL,
+		NULL,
+		PROV_RSA_FULL,
+		CRYPT_VERIFYCONTEXT))
+	{
+		goto error2;
+	}
+
+	if (!CryptCreateHash(hProv, CALG_MD5, 0, 0, &hHash)) {
+		goto error1;
+	}
+
+	if (!CryptHashData(hHash, data, len, 0)) {
+		goto error;
+	}
+
+	hash_len = MD5LEN;
+	if (!CryptGetHashParam(hHash, HP_HASHVAL, hash, &hash_len, 0)) {
+		goto error;
+	}
+
+	ret = TRUE;
+
+error:
+	CryptDestroyHash(hHash);
+error1:
+	CryptReleaseContext(hProv, 0);
+error2:
+
+	return ret;
+}
+
+#pragma warning (disable: 4996)
+
+wchar_t* hide_secret_str(PCWSTR lpData) {
+	DWORD len = 0;
+	if (lpData != NULL) {
+		len = (DWORD) wcslen(lpData);
+	}
+
+	return hide_secret(lpData, len);
+}
+
+wchar_t* hide_secret(LPCVOID lpData, DWORD cbData) {

```

Comment:
Add brief documentation for the function. What it does, mention the output format.

Reply:
done